database/database.go

```diff
@@ -0,0 +1,255 @@
+package database
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/NebulousLabs/skynet-accounts/build"
+	"github.com/NebulousLabs/skynet-accounts/user"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var (
+	// SkynetDBHostEV holds the name of the environment variable for DB host.
+	SkynetDBHostEV = "SKYNET_DB_HOST"
+	// SkynetDBPortEV holds the name of the environment variable for DB port.
+	SkynetDBPortEV = "SKYNET_DB_PORT"
+	// SkynetDBUserEV holds the name of the environment variable for DB username.
+	SkynetDBUserEV = "SKYNET_DB_USER"
+	// SkynetDBPassEV holds the name of the environment variable for DB password.
+	SkynetDBPassEV = "SKYNET_DB_PASS"
+
+	// SkynetDBName defines the name of Skynet's database.
+	SkynetDBName = "skynet"
+	// SkynetDBUsersCollection defines the name of the "users" collection within
+	// skynet's database.
+	SkynetDBUsersCollection = "users"
+
+	// ErrUserNotFound is returned when we can't find the user in question.
+	ErrUserNotFound = errors.New("user not found")
+	// ErrEmailAlreadyUsed is returned when we try to use an email to either
+	// either create or update a user and another user already uses this email.
+	ErrEmailAlreadyUsed = errors.New("email already in use by another user")
+	// ErrNoDBConnection is returned when we can't connect to the database.
+	ErrNoDBConnection = errors.New("no connection to the database")
+
+	// True is a helper, so we can easily provide a *bool for UpdateOptions.
+	True = true
+	// False is a helper, so we can easily provide a *bool for UpdateOptions.
+	False = false
+)
+
+// DB represents a MongoDB database connection.
+type DB struct {
+	ctx   context.Context
+	db    *mongo.Database
+	users *mongo.Collection
+}
+
+// New returns a new DB connection based on the environment variables.
+func New(ctx context.Context) (*DB, error) {
+	opts, err := connectionOptionsFromEnv()
+	if err != nil {
+		return nil, errors.AddContext(err, "failed to get all necessary connection parameters")
+	}
+	return NewCustom(ctx, opts[SkynetDBUserEV], opts[SkynetDBPassEV], opts[SkynetDBHostEV], opts[SkynetDBPortEV], SkynetDBName)
+}
+
+// NewCustom returns a new DB connection based on the passed parameters.
+func NewCustom(ctx context.Context, user, pass, host, port, dbname string) (*DB, error) {
+	connStr := fmt.Sprintf("mongodb://%s:%s@%s:%s/?compressors=zlib&gssapiServiceName=mongodb&readPreference=nearest", user, pass, host, port)
```

Comment:
Could be a helper. Since there are other params in there it might be worth making those constants so then the string creation is more clear as to what the inputs are.

Reply:
I'm not sure I completely follow.

Do you mean that the connection string can be moved to a helper method? I guess we can do that, I'm not 100% sure about the benefit if it's only used here.

Moving the other params to constants makes sense.